scripts/install_genproto_deps.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+#
+# Install all deps in correct version (matched to the vendored version) for ./genproto.sh to work.
+# This will forcibly checkout binaries in your current $GOPATH to match desired version as they are imported by
+# ./genproto.sh itself.
+
+set -e
+set -u
+
+if ! [[ "$0" =~ "scripts/install_genproto_deps.sh" ]]; then
+	echo "must be run from repository root"
+	exit 255
+fi
+
+DEPS=(
+"github.com/gogo/protobuf/protoc-gen-gogofast"
+"github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway"
+"github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger"
+"golang.org/x/tools/cmd/goimports"
+)
+
+for dep in ${DEPS[@]}
+do
+    # This regex helps to find "root" for Golang package which is always in form <url>/<author/org>/<package>
+    rootPackageRegex="([a-z.-]*/[a-z.-]*/[a-z.-]*)/.*"
+    if [[ "${dep}" =~ $rootPackageRegex ]]; then
+        rootPackage="${BASH_REMATCH[1]}"
+    else
+        rootPackage="${dep}"
+    fi
+
+    version=$(jq '[.package[] | select(.path|test("'"${rootPackage}"'."))][1].revision' ./vendor/vendor.json | tr -d '"')
```

Comment:
Do we require `jq` anywhere else in our builds today? If not it's technically a new dependency. I don't care myself, just noting it. If no one cares, I'm happy with merging with it.

Reply:
We don't, and it's not a standard tool.

Reply:
From my POV, this way of retrieving the package versions is brittle anyhow. FWIW etcd is pinning the exact dependencies in their `genproto.sh` script + get them in a scratch GOPATH directory. Although it is duplication from `vendor/vendor.json`, it is also more explicit and we don't update the protobuf dependencies that often.

https://github.com/etcd-io/etcd/blob/a18b46711125d08de37d0b71e9c0d42db363ed9e/scripts/genproto.sh#L21-L24

Reply:
I am fine to move to more explicit version, without auto deduction of deps version. Maybe instead we could add circle CI step to generate proto and compare with what is commited?